nsxt/resource_nsxt_vpc_subnet.go

```diff
@@ -359,8 +362,38 @@ func resourceNsxtVpcSubnetDelete(d *schema.ResourceData, m interface{}) error {
 	connector := getPolicyConnector(m)
 	parents := getVpcParentsFromContext(getSessionContext(d, m))
 
+	// Wait until potential VM ports are deleted
+	pendingStates := []string{"pending"}
+	targetStates := []string{"ok", "error"}
+	stateConf := &resource.StateChangeConf{
+		Pending: pendingStates,
+		Target:  targetStates,
+		Refresh: func() (interface{}, string, error) {
+			portsClient := subnets.NewPortsClient(connector)
+			ports, err := portsClient.List(parents[0], parents[1], parents[2], id, nil, nil, nil, nil, nil, nil)
+			if err != nil {
+				return ports, "error", logAPIError("Error listing VPC subnet ports", err)
```

Comment:
Should this fail because of an intermittent error in NSX, I guess Terraform will still retry invoking provider to perform the whole operation (until the number of retries expires, of course) ?

Reply:
Yes, regular http retry works here for status codes that are configured as retriable

Reply:
Got it, thanks!